### PR TITLE
karmada-operator: add CRDs archive verification to enhance file system robustness

### PR DESCRIPTION
**Pull request by zhzhuang-zju**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
`karmada-operator` provides a custom download path for CRDs files. Before processing the files, it's necessary to validate the CRDs archive to enhance system robustness.

- Check if its file name includes relative paths.

- check if the CRDs archive has the expected directory structure.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
`karamdactl init` has the same issue, and its enhancement is ongoing

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by zhzhuang-zju** [comment]
cc @RainbowMango

**Review by RainbowMango — COMMENTED** [review]
Just a nit, otherwise LGTM.

File: operator/pkg/tasks/init/crd.go [hunk 1]
```diff
@@ -184,3 +217,49 @@ func getCrdsDir(data InitData) (string, error) {
...
+		}
+	case tar.TypeReg:
+		if !isCleanPath(header.Name) {
+			return fmt.Errorf("the given file contains unclean file path: %s", header.Name)
+		}
+		if !isExpectedPath(header.Name, crdsArchive) {
+			return fmt.Errorf("the given file contains unexpected file path: %s", header.Name)
+		}
+	default:
+		fmt.Printf("unknown type: %v in %s\n", header.Typeflag, header.Name)
```
Comment (RainbowMango) [inline operator/pkg/tasks/init/crd.go:249]: This should be `unrelated file type`, which means we can tolerate that unrelated file in the tarball, right?

Comment (zhzhuang-zju) [reply operator/pkg/tasks/init/crd.go:249]: yes, we can reject this unintended `header.Type`

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve
